addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRuleTest.java

```diff
@@ -0,0 +1,328 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.NanoHTTPD.Response;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link SpringActuatorScanRule}. */
+class SpringActuatorScanRuleUnitTest extends ActiveScannerTest<SpringActuatorScanRule> {
+
+    private static final String ACTUATOR_CONTENT_TYPE =
+            "Content-Type: application/vnd.spring-boot.actuator.v2+json;charset=UTF-8";
+    private static final String RELEVANT_BODY = "{\"status\":\"UP\"}";
+    private static final String IRRELEVANT_BODY = "<body><h1>Howdy></h1></body>";
+    private static final String REQ_PATH = "actuator/health";
+
+    @Override
+    protected SpringActuatorScanRule createScanner() {
+        return new SpringActuatorScanRule();
+    }
+
+    @Override
+    protected void setUpMessages() {
+        mockMessages(new ExtensionAscanRulesAlpha());
+    }
+
+    @Test
+    void shouldTargetJavaTech() {
+        // Given
+        TechSet techSet = techSet(Tech.JAVA);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotTargetNonJavaTechs() {
+        // Given
+        TechSet techSet = techSetWithout(Tech.JAVA);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(false)));
+    }
+
+    @Test
+    void shouldBeInfoGatherAlert() {
+        // Given
+        Integer category = Category.INFO_GATHER;
+        // When
+        Integer ruleCategory = rule.getCategory();
+        // Then
+        assertEquals(ruleCategory, category);
+    }
+
+    @Test
+    void shouldSkipScanIfStopped() throws Exception {
+        // Given
+        rule.init(getHttpMessage("/"), parent);
+        parent.stop();
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, is(empty()));
+    }
```

Comment:
It works but it's pointless, a stopped scan does not run the scan rules. To properly test the behaviour the scan rules should be stopped while running (in this case it would show that it requires the stop in the inner loop too).

Reply:
Fair enough. I'll remove. I think it does actually test the function in the rule, but I borrowed heavily from the HiddenFile scanner :)

Reply:
It does cover that stop check statement, but that one is not actually relevant (yet), the loop just has one element (the inner loop has two, which should check for stop as well). I'd change the test to stop when the first request is received and then assert that no other request is sent, that would match better the more likely scenario.

Reply:
@thc202 I could really use a hint on how to do this here. I'm not sure I know how to make a scan start and then in the test shut it down after 1 message is sent.. rule.scan(); doesn't come back until it's done (AFAIK) - I tried some stuff, but I'm kind of stuck

Reply:
Add a server handler that stops the scan when serving the request.

Reply:
I have no idea how to make this work correctly since it doesn't seem like nano has the ability to serve one then do a thing. Happy to put the old check back, but I don't have more time to spend on this check so I removed